eng/tools/TypeSpecValidation/src/TypeSpecValidation.ts

```diff
@@ -48,6 +48,17 @@ export async function main() {
   const folder = parsedArgs.positionals[0];
   console.log("Running TypeSpecValidation on folder:", folder);
 
+  // Verify specs' path requirements
+  let leafDir = folder.substring(folder.lastIndexOf(path.sep) + 1);
+  if (leafDir.includes(".") && !/\w\.[A-Z]/.test(leafDir)) {
+    throw new Error(
+      "Please rename the folder " +
+        leafDir +
+        " to " +
+        leafDir.replace(/\.\w/g, (l) => l.toUpperCase())
```

Comment:
Won't this also cause any other of the same letters to be upper cased in the path?

Reply:
@weshaggard: I converted this PR back to draft.  It will have significant changes before merging.